app/controllers/research_sessions_controller.rb

```diff
@@ -30,7 +30,7 @@ def preview
       @research_session.assign_attributes(params.permit(:age))
     end
 
-    @research_session = ResearchSessionPresenter.new(@research_session)
+    @preview = ResearchSessionPresenter.new(@research_session)
```

Comment:
Rails-y suggestion: leave this as `@research_session` (don't add a separate `@preview` field) and don't check for the existence of `@preview`. Parameterise the _progress partial instead. At https://github.com/barnardos/consent-form-builder-rails/pull/81/files#diff-6e7794c2e7657c77f69b1250c49e3774R1 , say

```ruby
<%= render "research_sessions/progress", locals: { preview: true } %>
```

and in the _progress partial itself, default `preview` to `false` by saying

`<% preview ||= false %>`

Then elsewhere in _progress you'll have a local variable preview which you can use as a variable local to that partial:

```
 <li class="progress__stage <%= 'is-active' if preview %>">
```

Lastly, add `:reached_step?` to the delegate list in the presenter.

Reply:
it was left as research session before and I tried adding reached_step? to it but it was not having it. It's maybe due to steps and session both having a reached_step method (think that may need work).

I'll revert it to what you are suggesting to demonstrate it failing when I come in.

Reply:
actually doing it that way means I don't need to extend the delegate, but it did feel a bit wrong to put it there.

Reply:
My original reason for not passing a param and looking at session state was it feels like it's mixing having logic built into the fragment that looks at state and having parameters sent into it, ideally it would be all passed as params